packages/lib/src/components/ThreeDS2/components/Challenge/types.ts

```diff
@@ -18,10 +18,16 @@ export interface DoChallenge3DS2State {
 export interface PrepareChallenge3DS2Props extends ThreeDS2ChallengeConfiguration {
     onComplete?: (data: ChallengeResolveData) => void;
     onSubmitAnalytics: (aObj: SendAnalyticsObject) => void;
+    isMDFlow: boolean;
 }
 
 export interface PrepareChallenge3DS2State {
-    challengeData?: ChallengeData;
+    challengeData?: ChallengeData | ErrorObject;
     status?: string;
```

Comment:
I think we could even narrow down the status to `status?: 'performingChallenge' | 'error' | 'init' | 'complete';`

Reply:
Yes, indeed